exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/metric_exporter/__init__.py

```diff
@@ -202,7 +207,85 @@ def export(
         **kwargs,
     ) -> MetricExportResult:
         # TODO(#2663): OTLPExporterMixin should pass timeout to gRPC
-        return self._export(metrics_data)
+        if not self.max_export_batch_size:
+            return self._export(data=metrics_data)
+
+        if len(metrics_data) <= self.max_export_batch_size:
+            return self._export(data=metrics_data)
+
+        for split_metrics_data in self._split_metrics_data(metrics_data):
+            export_result: MetricExportResult = self._export(
+                data=split_metrics_data
+            )
+            if export_result is MetricExportResult.FAILURE:
+                return export_result
+
+        return MetricExportResult.SUCCESS
+
+    def _split_metrics_data(
+        self,
+        metrics_data: MetricsData,
+    ) -> Iterable[MetricsData]:
+        for index in range(0, len(metrics_data), self.max_export_batch_size):
+            yield self.__split_metrics_data(
+                start_metric_index=index,
+                stop_metric_index=index + self.max_export_batch_size,
+                metrics_data=metrics_data,
+            )
+
+    def __split_metrics_data(
```

Comment:
[Double leading underscores](https://docs.python.org/3/tutorial/classes.html#private-variables) in class attributes have a special meaning in Python. Please use another name if this is not your intention. Also consider moving the code in `__split_metrics_data` directly in `_split_metrics_data` since it is a single use function.